File: 2024-spring-examples/lecture02-counters/counters.py
incrementBy = 1

def makeCounter(name):
	counter = 0
	def tmp():
		nonlocal counter
		global incrementBy
		print(f"{name} incrementBy: {incrementBy} -> {incrementBy + 1}")
		incrementBy += 1
		print(f"{name} counter: {counter} -> {counter + incrementBy}")
		counter += incrementBy
	return tmp

counterA = makeCounter("a")
counterB = makeCounter("b")

print("counterA()")
counterA()
print("\ncounterB()")
counterB()
print("\ncounterA()")
counterA()
print("\ncounterB()")
counterB()

File: 2024-spring-examples/lecture02-counters/counters.js
let incrementBy = 1

function makeCounter(name) {
	let counter = 0
	return () => {
		console.log(`${name} incrementBy: ${incrementBy} -> ${incrementBy + 1}`)
		++incrementBy
		console.log(`${name} counter: ${counter} -> ${counter + incrementBy}`)
		counter += incrementBy
	}
}

const counterA = makeCounter("a")
const counterB = makeCounter("b")

console.log("counterA()")
counterA()
console.log("\ncounterB()")
counterB()
console.log("\ncounterA()")
counterA()
console.log("\ncounterB()")
counterB()
